swap_meet/vendor.py

```diff
@@ -1,2 +1,58 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, new_item):
+        self.inventory.append(new_item)
+        return new_item
+
+    def remove(self, item_to_remove):
+        if item_to_remove in self.inventory:
+            removed_item = self.inventory.pop(self.inventory.index(item_to_remove))
+            return removed_item
+        return False
+
+    def get_by_category(self, category):
+        category_items = []
+        for item in self.inventory:
+            if item.category == category:
+                category_items.append(item)
+        return category_items
+
+    def swap_items(self, other, my_item, their_item):
+        if my_item in self.inventory and their_item in other.inventory:
+            # Pops each item from their respective inventories and swaps them using Vendor.add()
+            other.add(self.inventory.pop(self.inventory.index(my_item)))
+            self.add(other.inventory.pop(other.inventory.index(their_item)))
+            return True
+        return False
```

Comment:
This works well! A slightly more OOP way to do it would be to use the `add` and `remove` methods on the `Vendor` class. That way if we ever add any additional action to those functions (maybe something like a counter that shows how many times an item has ever been removed from the Vendor's inventory), we get that functionality for free here too. It also allows any potential subclass of `Vendor` to just change the `add` and `remove` to change that functionality without needing to also override `swap_items`.